src/circleci/rollcage/json.clj

```diff
@@ -30,13 +30,40 @@
      (catch JsonGenerationException _
        (str item)))))
 
+(defn- fields-to-scrub
+  [block-fields]
+  (-> block-fields
+      (concat (map snake-case block-fields))
+      (concat (map name block-fields))
+      (concat (map (comp keyword snake-case) block-fields))))
+
+(defn- remove-fields-from-map [a-map block-fields]
+  (reduce
+   (fn remove-field [acc field]
+     (if (get acc field)
+       (assoc acc field "*field removed*")
+       acc))
+   a-map
+   block-fields))
+
+(defn- remove-fields [item block-fields]
+  (if (seq block-fields)
+    (postwalk
+     (fn [form]
+       (if (map? form)
+         (remove-fields-from-map form block-fields)
+         form))
+     item)
+    item))
+
 (defn encode
   "Convert an arbitrary object into a JSON string."
-  [item]
-  (try
-   (json/generate-string item {:key-fn snake-case})
-   (catch JsonGenerationException _
-     (json/generate-string (postwalk backstop-encoder item) {:key-fn snake-case}))))
+  [item block-fields]
```

Comment:
Since this is a public function, maybe we should include an arity where `block-fields` may be omitted so it can default to an empty list?